packages/logger/tests/unit/injectLambdaContext.test.ts

```diff
@@ -239,4 +240,339 @@ describe('Inject Lambda Context', () => {
+      // Prepare
+      const logger = new Logger(); // No search function provided
+      const warnSpy = vi.spyOn(logger, 'warn');
+
+      // Act - Use middleware which will internally call setCorrelationIdFromPath
+      const handler = middy(async () => {
+        logger.info('Hello, world!');
+      }).use(
+        injectLambdaContext(logger, {
+          correlationIdPath: 'headers.x-correlation-id',
+        })
+      );
+
+      await handler(testEvent, context);
+
+      // Assess
+      expect(warnSpy).toHaveBeenCalledWith(
+        'correlationIdPath is set but no search function was provided. The correlation ID will not be added to the log attributes.'
+      );
+    });
+
+    it('should not set correlation ID when search function returns falsy value', async () => {
+      // Prepare
+      const logger = new Logger({ correlationIdSearchFn: search });
+
+      // Act - Use middleware which will internally call setCorrelationIdFromPath
+      const handler = middy(async () => {
+        logger.info('Hello, world!');
+      }).use(
+        injectLambdaContext(logger, {
+          correlationIdPath: 'headers."non-existent"',
+        })
+      );
+
+      await handler(testEvent, context);
+
+      // Assess
+      expect(logger.getCorrelationId()).toBeUndefined();
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.not.objectContaining({
+          correlation_id: expect.anything(),
+        })
+      );
+    });
+
+    it('should propagate correlation ID to child loggers', () => {
+      // Prepare
+      const logger = new Logger();
+      logger.setCorrelationId('parent-correlation-id');
+
+      // Act
+      const childLogger = logger.createChild();
+      childLogger.info('Hello from child!');
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.objectContaining({
+          message: 'Hello from child!',
+          correlation_id: 'parent-correlation-id',
+        })
+      );
+    });
+
+    it('should allow using different types as correlation ID', () => {
+      // Prepare
+      const logger = new Logger();
+      const numericId = 12345;
+
+      // Act
+      logger.setCorrelationId(numericId);
+      logger.info('Using numeric ID');
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.objectContaining({
+          message: 'Using numeric ID',
+          correlation_id: numericId,
+        })
+      );
+      expect(logger.getCorrelationId()).toBe(numericId);
+    });
+
+    it('should handle complex objects as correlation ID', () => {
+      // Prepare
+      const logger = new Logger();
+      const complexId = { requestId: 'abc123', timestamp: Date.now() };
+
+      // Act
+      logger.setCorrelationId(complexId);
```

Comment:
At least with the current implementation - this is just an alias of set attribute, the test does nothing.